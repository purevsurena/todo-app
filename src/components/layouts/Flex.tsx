import React from "react";
import { View, StyleSheet, ViewStyle } from "react-native";

interface FlexProps {
  flex?: number;
  flexDirection?: ViewStyle["flexDirection"];
  justifyContent?: ViewStyle["justifyContent"];
  alignItems?: ViewStyle["alignItems"];
  alignSelf?: ViewStyle["alignSelf"];
  style?: ViewStyle;
  children?: React.ReactNode;
  testID?: string;
}

const Flex: React.FC<FlexProps> = ({
  flex,
  flexDirection,
  justifyContent,
  alignItems,
  alignSelf,
  style,
  children,
}) => {
  const flexStyle: ViewStyle = {
    flex,
    flexDirection,
    justifyContent,
    alignItems,
    alignSelf,
  };

  return (
    <View testID="flexComponent" style={[styles.flex, flexStyle, style]}>
      {children}
    </View>
  );
};

const styles = StyleSheet.create({
  flex: {
    display: "flex",
  },
});

export default Flex;
