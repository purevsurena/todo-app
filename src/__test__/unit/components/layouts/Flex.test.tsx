/* eslint-disable no-undef */
import React from "react";
import { render } from "@testing-library/react-native";
import Flex from "@/components/layouts/Flex";
import { Text } from "react-native";

describe("Flex Component", () => {
  it("should render children correctly", () => {
    const { getByText } = render(
      <Flex>
        <Text>Test Child</Text>
      </Flex>,
    );
    expect(getByText("Test Child")).toBeTruthy();
  });

  it("should apply the flex style correctly", () => {
    const { getByTestId } = render(<Flex flex={1} testID="flexComponent" />);
    const flexComponent = getByTestId("flexComponent");
    expect(flexComponent.props.style).toContainEqual({ flex: 1 });
  });

  it("should merge style prop correctly", () => {
    const { getByTestId } = render(
      <Flex style={{ backgroundColor: "red" }} testID="flexComponent" />,
    );
    const flexComponent = getByTestId("flexComponent");
    expect(flexComponent.props.style).toContainEqual({
      backgroundColor: "red",
    });
  });

  it("matches the snapshot", () => {
    const { toJSON } = render(
      <Flex
        flex={1}
        flexDirection="row"
        justifyContent="center"
        alignItems="center"
        style={{ backgroundColor: "blue" }}
      >
        <Text>Snapshot Test</Text>
      </Flex>,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
