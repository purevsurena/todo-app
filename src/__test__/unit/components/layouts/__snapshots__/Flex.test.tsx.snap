// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flex Component matches the snapshot 1`] = `
<View
  style={
    [
      {
        "display": "flex",
      },
      {
        "alignItems": "center",
        "alignSelf": undefined,
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
      },
      {
        "backgroundColor": "blue",
      },
    ]
  }
  testID="flexComponent"
>
  <Text>
    Snapshot Test
  </Text>
</View>
`;
